packages/service-library/src/servicelib/long_running_tasks/_task.py

```diff
@@ -104,51 +115,49 @@ async def _stale_tasks_monitor_worker(self) -> None:
                 logger.warning(
                     "Removing stale task '%s' with status '%s'",
                     task_id,
-                    self.get_status(task_id).json(),
+                    self.get_task_status(task_id).json(),
                 )
-                await self.remove(task_id, reraise_errors=False)
+                await self.remove_task(task_id, reraise_errors=False)
 
     @staticmethod
-    def get_task_id(task_name: TaskName) -> str:
+    def _create_task_id(task_name: TaskName) -> str:
         return f"{task_name}.{uuid4()}"
 
-    def is_task_name_running(self, task_name: TaskName) -> bool:
+    def is_task_running(self, task_name: TaskName) -> bool:
         """returns True if a task named `task_name` is running"""
-        if task_name not in self.tasks:
+        if task_name not in self._tasks_groups:
             return False
 
-        managed_tasks_ids = list(self.tasks[task_name].keys())
+        managed_tasks_ids = list(self._tasks_groups[task_name].keys())
         return len(managed_tasks_ids) > 0
 
-    def add(
+    def add_task(
         self, task_name: TaskName, task: Task, task_progress: TaskProgress
     ) -> TrackedTask:
-        task_id = self.get_task_id(task_name)
+        task_id = self._create_task_id(task_name)
 
-        if task_name not in self.tasks:
-            self.tasks[task_name] = {}
+        if task_name not in self._tasks_groups:
+            self._tasks_groups[task_name] = {}
 
-        tracked_task = TrackedTask.parse_obj(
-            dict(
-                task_id=task_id,
-                task=task,
-                task_name=task_name,
-                task_progress=task_progress,
-            )
+        tracked_task = TrackedTask(
```

Comment:
So if I get this correct, you bypass validators here?

Reply:
Why would validator be bypassed?? This is the standard way to construct a pydantic object
``parse_obj`` is when you receive some sort dict/list object and there is an extra logic (check the implementation of ``parse_obj``

SEE the examples in https://pydantic-docs.helpmanual.io/usage/validators/

Reply:
I saw something related to validators being bypassed when the constructor is used. Not sure exactly what was happening since some time passed since then. But yes, it makes sense that all validators would work on the constructor and not just when parsing.